hazelcast/src/main/java/com/hazelcast/config/WanReplicationRef.java

```diff
@@ -141,6 +159,10 @@ public void writeData(ObjectDataOutput out) throws IOException {
             out.writeUTF(filter);
         }
         out.writeBoolean(republishingEnabled);
+        // RU_COMPAT_3_10
```

Comment:
Does the class need to be `Versioned` now? Or is this covered by `MapConfig`?
Also, this class is annotated with `@BinaryInterface`. We might need to configure it outside this class :(

Reply:
By looking at the references, I'm guessing the annotation comes from the fact that it's used in `com.hazelcast.config.CacheConfig#wanReplicationRef`, which is annotated with `@BinaryInterface` itself.

Reply:
added Versioned.

Reply:
But from the javadoc of `BinaryInterface`:
```
 * Indicates that the binary serialized form of the annotated class is used in client-member communication over Hazelcast
 * Open Binary Client Protocol or in other ways which may break Hazelcast or other systems. Changing the way
 * {@code @BinaryInterface} classes are serialized will result in incompatibilities, so the following rules
 * apply to classes annotated with this annotation in the 3.x release series:
 * <p>
 * - NEVER CHANGE THEM<br>
 * - NEVER MAKE THEM IMPLEMENT THE VERSIONED INTERFACE
```